engine/table/src/main/java/io/deephaven/engine/table/impl/sources/UnionSourceManager.java

```diff
@@ -556,8 +558,13 @@ private void processExisting(@NotNull final Table constituent) {
                     resultRows.insertWithShift(currFirstRowKey, constituent.getRowSet());
                 }
             } else if (shiftDelta != 0) {
```

Comment:
Please look at L534:
`final boolean needToProcessShifts = changes.shifted().nonempty() && constituent.getRowSet().isNonempty();`
Maybe we should also be checking that previous was non-empty (using `sizePrev()`)?

Reply:
Seems like a reasonable condition to add; shifts are expensive.